### PR TITLE
Use OSS Licenses Plugin to display license under "View Licenses" in menu

### LINKED ISSUE #2326: [Compliance] Display licenses and ToS in app
Potentially using https://github.com/google/play-services-plugins/tree/master/oss-licenses-plugin

### PR DESCRIPTION
**Pull request by sufyanAbbasi**
Fixes #2326 

Introduces the [oss-licenses-plugin](https://github.com/google/play-services-plugins/tree/master/oss-licenses-plugin) that builds the licenses for each dependency and shows them in a dynamically generated list.

- [x] Verified that "View Licenses" button triggers the plugin in debug mode.
- [x] Verified that licenses are shown in release build.

"View Licenses":
Screenshot 2024-03-26 at 2 00 39 PM

Debug build:
Screenshot 2024-03-26 at 1 54 06 PM

Release build:
Screenshot 2024-03-26 at 1 58 29 PM
Screenshot 2024-03-26 at 1 58 39 PM
Screenshot 2024-03-26 at 1 58 47 PM
Some are inlined:
Screenshot 2024-03-26 at 1 59 46 PM

PTAL @gino-m!

### CODE REVIEW

File: ground/build.gradle [hunk 1]
```diff
@@ -267,6 +269,9 @@ dependencies {
     implementation "androidx.work:work-runtime-ktx:$workVersion"
     testImplementation "androidx.work:work-testing:$workVersion"
 
+    // OSS Licenses Plugin
+    implementation 'com.google.android.gms:play-services-oss-licenses:17.0.1'
```
Comment (gino-m) [inline ground/build.gradle:273]: Nice find!

**Comment by gino-m** [comment]
Hi @sufyanAbbasi! Did you manage to build a release APK to test the licenses?

**Comment by sufyanAbbasi** [comment]
> Hi @sufyanAbbasi! Did you manage to build a release APK to test the licenses?

Yes I did, thanks to @shobhitagarwal1612's help. Please see the screenshots. Note that it just borrows the app theme, which is currently set to `AppTheme.Launcher`, hence the logo in the middle of the page.

**Comment by gino-m** [comment]
@sufyanAbbasi Lmk once you fix the theme (icon showing in the middle of the screen) and we can merge.

File: ground/build.gradle (hunk 1 above)
Comment (sufyanAbbasi) [reply ground/build.gradle:272]: By your suggestion! :)

**Comment by sufyanAbbasi** [comment]
> @sufyanAbbasi Lmk once you fix the theme (icon showing in the middle of the screen) and we can merge. We can move the link to the licenses in our "About" page once designs and impl are ready.

Fixed now... but I had to change the entire App theme from `AppTheme.Launcher` to `AppTheme.Base`. I didn't detect any diffs when I did this!

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt [hunk 2]
```diff
@@ -71,6 +73,7 @@ class HomeScreenFragment :
       ->
       onApplyWindowInsets(insets)
     }
+    OssLicensesMenuActivity.setActivityTitle(getString(R.string.view_licenses_title))
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt:76]: Can we move this to `showLicenses` method to keep all related code together?

**Comment by shobhitagarwal1612** [comment]
Please see https://medium.com/@jokatavr/how-to-use-android-open-source-notices-299a0635b5c2

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt (hunk 2 above)
Comment (sufyanAbbasi) [reply ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt]: Done.

**Comment by sufyanAbbasi** [comment]
Just fixed the build issue and updated against main.

**Comment by shobhitagarwal1612** [comment]
Can you please verify that the splash logo is still visible when opening the app?

**Comment by sufyanAbbasi** [comment]
> Can you please verify that the splash logo is still visible when opening the app?

On Emulator: 
Screenshot 2024-04-04 at 4 08 55 PM 3

On Pixel (dark theme applied):
share_2355828271026720996

**Review by shobhitagarwal1612 — APPROVED** [review]

**Pull request merged**